store.py

```diff
@@ -22,65 +22,182 @@
+            i += 1
+        while i < (self._regular_count + self._express_count):
+            self._line_list.append(ExpressLine(self._line_capacity, True, []))
+            i += 1
+        while i < (self._regular_count + self._express_count
+                   + self._self_serve_count):
+            self._line_list.append(SelfServeLine(self._line_capacity, True, []))
+            i += 1
+
+    def get_info(self, name: str) -> int:
+        """Return requested info from input file. Allows indirect access to
+        private attributes.
+        Primarily to be used in doctests and pytests
+        >>> import io
+        >>> config_file = io.StringIO('{"regular_count":0,"express_count":0,"self_serve_count":1,"line_capacity":10}')
+        >>> g = GroceryStore(config_file)
+        >>> g.get_info('_regular_count')
+        0
+        >>> g.get_info('_express_count')
+        0
+        >>> g.get_info('_self_serve_count')
+        1
+        >>> g.get_info('_line_capacity')
+        10
+        """
+
+        if name == '_regular_count':
+            return self._regular_count
+        elif name == '_express_count':
+            return self._express_count
+        elif name == '_self_serve_count':
+            return self._self_serve_count
+        elif name == '_line_capacity':
+            return self._line_capacity
+        else:
+            return 0
+
+
+    def get_line_list(self) -> List:
+        """Return the list of lines in this
+
         """
+        result = []
+        for line in self._line_list:
+            result.append(line)
+        return result
 
     def enter_line(self, customer: Customer) -> int:
         """Pick a new line for <customer> to join.
 
         Return the index of the line that the customer joined.
         Must use the algorithm from the handout.
 
+        When a new customer arrives, they join the open line with the fewest
+        number of customers that they are allowed to join, choosing the one
+        with the lowest index (as represented by the grocery store)
+        if there is a tie.
+
+        When there are no lines the customer can join, the “new customer” event
+        should go back into the container, and have its timestamp increased by
+        1 (representing trying to join a line again at the next time interval.)
+
         Return -1 if there is no line available for the customer to join.
+        >>> import io
+        >>> config_file = io.StringIO('{"regular_count":0,"express_count":0,"self_serve_count":1,"line_capacity":10}')
+        >>> g = GroceryStore(config_file)
+        >>> customer = Customer('bill nye', [Item('banana', 5), Item('apple', 6)])
+        >>> g.enter_line(customer)
+        0
         """
 
+        i = -1
+        for line in self._line_list:
+            if line.can_accept(customer):
+                line.queue.append(customer)
+                break
+            else:
+                i += 1
```

Comment:
this mechanism does not account for the condition where the customer would join the shortest line